src/main/java/us/tastybento/bskyblock/Settings2.java

```diff
@@ -0,0 +1,1115 @@
+
+    private boolean makeIslandIfNone;
+    private boolean immediateTeleportOnIsland;
+    private boolean respawnOnIsland;
+
+    // Deaths
+    @ConfigEntry(path = "island.deaths.max")
+    private int deathsMax = 10;
+
+    @ConfigEntry(path = "island.deaths.sum-team")
+    private boolean deathsSumTeam = false;
+
+    // ---------------------------------------------
+
+    /*      PROTECTION      */
+    @ConfigEntry(path = "protection.allow-piston-push")
+    private boolean allowPistonPush = false;
+
+    @ConfigEntry(path = "protection.restrict-flying-mobs")
+    private boolean restrictFlyingMobs = true;
+
+    private int togglePvPCooldown;
+
+    private HashMap<Flag, Boolean> defaultFlags;
+
+    //TODO transform these options below into flags
+    private boolean allowEndermanGriefing;
+    private boolean endermanDeathDrop;
+    private boolean allowTNTDamage;
+    private boolean allowChestDamage;
+    private boolean allowCreeperDamage;
+    private boolean allowCreeperGriefing;
+    private boolean allowMobDamageToItemFrames;
+
+    //TODO flags
+
+    // ---------------------------------------------
+
+    /*      ACID        */
+
+    /*
+     * This settings category only exists if the GameType is ACIDISLAND.
+     */
+
+    @ConfigEntry(path = "acid.damage-op", specificTo = GameType.ACIDISLAND)
+    private boolean acidDamageOp = false;
+
+    @ConfigEntry(path = "acid.damage-chickens", specificTo = GameType.ACIDISLAND)
+    private boolean acidDamageChickens = false;
+
+    @ConfigEntry(path = "acid.options.item-destroy-time", specificTo = GameType.ACIDISLAND)
+    private int acidDestroyItemTime = 0;
+
+    // Damage
+    @ConfigEntry(path = "acid.damage.acid.player", specificTo = GameType.ACIDISLAND)
+    private int acidDamage = 10;
+
+    @ConfigEntry(path = "acid.damage.rain", specificTo = GameType.ACIDISLAND)
+    private int acidRainDamage = 1;
+
+    @ConfigEntry(path = "acid.damage.effects", specificTo = GameType.ACIDISLAND)
+    private List<PotionEffectType> acidEffects = new ArrayList<>(Arrays.asList(PotionEffectType.CONFUSION, PotionEffectType.SLOW));
+
+    /*      SCHEMATICS      */
+    private List<String> companionNames;
+    private ItemStack[] chestItems;
+    private EntityType companionType;
+
+    private boolean useOwnGenerator;
+
+    private HashMap<String,Integer> limitedBlocks;
+    private boolean teamJoinDeathReset;
+    
+    private String uniqueId;
+
+    /**
+     * @return the uniqueId
+     */
+    public String getUniqueId() {
+        return uniqueId;
+    }
+    /**
+     * @param uniqueId the uniqueId to set
+     */
+    public void setUniqueId(String uniqueId) {
+        this.uniqueId = uniqueId;
+    }
+    /**
+     * @return the checkUpdates
+     */
+    public boolean isCheckUpdates() {
+        return checkUpdates;
+    }
+    /**
+     * @param checkUpdates the checkUpdates to set
+     */
+    public void setCheckUpdates(boolean checkUpdates) {
```

Comment:
Having these methods public may result in developers changing them on the fly. Which might be really, really dangerous.